time_tracker_api/time_entries/time_entries_repository.py

```diff
@@ -233,6 +235,48 @@ def find_all_v2(
         function_mapper = self.get_mapper_or_dict(mapper)
         return list(map(function_mapper, result))
 
+    def get_last_entry(
+        self,
+        owner_id: str,
+        event_context: EventContext,
+        visible_only=True,
+        mapper: Callable = None,
+    ):
+        query_builder = (
+            CosmosDBQueryBuilder()
+            .add_sql_where_equal_condition({'owner_id': owner_id})
+            .add_sql_order_by_condition('end_date', Order.DESC)
+            .add_sql_limit_condition(1)
+            .add_sql_offset_condition(1)
+            .build()
+        )
+
+        query_str = query_builder.get_query()
+        params = query_builder.get_parameters()
+
+        partition_key_value = self.find_partition_key_value(event_context)
+        result = self.container.query_items(
+            query=query_str,
+            parameters=params,
+            partition_key=partition_key_value,
+        )
+
+        function_mapper = self.get_mapper_or_dict(mapper)
+        return function_mapper(next(result))
+
+    def update_last_entry(
+        self, owner_id, start_date, event_context: EventContext
+    ):
+        last_entry = self.get_last_entry(owner_id, event_context)
+        start_date_tmp = start_date
+
+        end_date = str_to_datetime(last_entry.end_date)
+        start_date = str_to_datetime(start_date)
```

Comment:
I think you meant `str_to_datetime(start_date_tmp)`.

Reply:
No, start_date or start_date_tmp it's ok, but with line 274 I convert the string in a DateTime object, so it causes an error if a pass star_date in the update_data because it is not a string, I create the start_dte_tmp to keep that string

Reply:
You can use `_start_date` variable name, so you avoid the use of `start_date_tmp` . Choose what it works best for you.